sql/moz-fx-data-marketing-prod/acoustic/contact_raw_v1/query.sql

```diff
@@ -0,0 +1,76 @@
+    FORMAT_FXA_DATE(fxa_created_date) AS fxa_created_date,
+    LOWER(COALESCE(NULLIF(fxa_first_service, ''), "unknown")) AS fxa_first_service,
+    CAST(COALESCE(CAST(fxa_account_deleted AS INTEGER), 0) AS BOOLEAN) AS fxa_account_deleted,
+    CAST(COALESCE(CAST(sub_mozilla_foundation AS INTEGER), 0) AS BOOLEAN) AS sub_mozilla_foundation,
+    CAST(COALESCE(CAST(sub_common_voice AS INTEGER), 0) AS BOOLEAN) AS sub_common_voice,
+    CAST(COALESCE(CAST(sub_hubs AS INTEGER), 0) AS BOOLEAN) AS sub_hubs,
+    CAST(COALESCE(CAST(sub_mixed_reality AS INTEGER), 0) AS BOOLEAN) AS sub_mixed_reality,
+    CAST(
+      COALESCE(CAST(sub_internet_health_report AS INTEGER), 0) AS BOOLEAN
+    ) AS sub_internet_health_report,
+    CAST(COALESCE(CAST(sub_miti AS INTEGER), 0) AS BOOLEAN) AS sub_miti,
+    CAST(
+      COALESCE(CAST(sub_mozilla_fellowship_awardee_alumni AS INTEGER), 0) AS BOOLEAN
+    ) AS sub_mozilla_fellowship_awardee_alumni,
+    CAST(COALESCE(CAST(sub_mozilla_festival AS INTEGER), 0) AS BOOLEAN) AS sub_mozilla_festival,
+    CAST(COALESCE(CAST(sub_mozilla_technology AS INTEGER), 0) AS BOOLEAN) AS sub_mozilla_technology,
+    CAST(COALESCE(CAST(sub_mozillians_nda AS INTEGER), 0) AS BOOLEAN) AS sub_mozillians_nda,
+    CAST(
+      COALESCE(CAST(sub_firefox_accounts_journey AS INTEGER), 0) AS BOOLEAN
+    ) AS sub_firefox_accounts_journey,
+    CAST(COALESCE(CAST(sub_knowledge_is_power AS INTEGER), 0) AS BOOLEAN) AS sub_knowledge_is_power,
+    CAST(
+      COALESCE(CAST(sub_take_action_for_the_internet AS INTEGER), 0) AS BOOLEAN
+    ) AS sub_take_action_for_the_internet,
+    CAST(COALESCE(CAST(sub_test_pilot AS INTEGER), 0) AS BOOLEAN) AS sub_test_pilot,
+    CAST(COALESCE(CAST(sub_firefox_news AS INTEGER), 0) AS BOOLEAN) AS sub_firefox_news,
+    CAST(COALESCE(CAST(sub_about_mozilla AS INTEGER), 0) AS BOOLEAN) AS sub_about_mozilla,
+    CAST(COALESCE(CAST(sub_apps_and_hacks AS INTEGER), 0) AS BOOLEAN) AS sub_apps_and_hacks,
+    CAST(COALESCE(CAST(sub_rally AS INTEGER), 0) AS BOOLEAN) AS sub_rally,
+    CAST(
+      COALESCE(CAST(NULLIF(sub_firefox_sweepstakes, '') AS INTEGER), 0) AS BOOLEAN
+    ) AS sub_firefox_sweepstakes,
+    LOWER(NULLIF(vpn_waitlist_geo, '')) AS vpn_waitlist_geo,
+    SPLIT(NULLIF(vpn_waitlist_platform, ''), ",") AS vpn_waitlist_platform,
+    LOWER(NULLIF(relay_waitlist_geo, '')),
+    RECIPIENT_ID AS recipient_id,
+    DATE(@submission_date) AS submission_date,
+  FROM
+    `moz-fx-data-bq-fivetran.acoustic_sftp.contact_export`
+  -- The table created by Fivetran is not partitioned, and the data is not always great in quality
+  -- the data loaded into the Fivetran table is cleaned up and then loaded into a downstream table.
+  -- Once the data is cleaned and loaded in to the target table (partioning and cluster fields enabled)
+  -- this temporary/staging table (data) is no longer needed.
+    TRUNCATE TABLE `moz-fx-data-bq-fivetran.acoustic_sftp.contact_export`;
+END;
```

Comment:
Discussed this a bit offline. I'm still unsure about the pattern of truncating the source though.. A couple of  questions/thoughts:
1. As it is now rebuilding this table means re-running the connector from the beginning of time instead of just the queries. It would be nice to break this dependency.. 
2. How does this interact with Fivetran billing? If I understood correctly from your description the connector only returns changed entries so the net Monthly Active Rows is the same?
3. Is it possible to add partitioning to these tables somehow? If not in the [fivetran native way](https://fivetran.com/docs/destinations/bigquery/partition-table), potentially through an intermediate append-only table?

Reply:
okay, I modified the workflow a bit. Now we create a raw table that is partitioned by submission date which then is used when creating a downstream table with the data being processed.